File: sponge-nashorn/examples/script/js/triggers_event_pattern_incorrect.js
/**
 * Sponge Knowledge Base
 * Triggers - Incorrect event pattern
 */

var TriggerAPattern = Java.extend(Trigger, {
    onConfigure: function(self) {
        self.withEvent("a.**");
    },
    onRun: function(self, event) {
    }
});

File: sponge-kotlin/src/test/kotlin/examples/TriggersEventPatternIncorrect.kt
/*
 * Sponge Knowledge Base
 * Triggers - Incorrect event pattern
 */

package org.openksavi.sponge.kotlin.examples

import org.openksavi.sponge.event.Event
import org.openksavi.sponge.kotlin.KKnowledgeBase
import org.openksavi.sponge.kotlin.KTrigger

class TriggersEventPatternIncorrect : KKnowledgeBase() {

    class TriggerAPattern : KTrigger() {
        override fun onConfigure() {
            withEvent("a.**")
        }
        override fun onRun(event: Event) {
        }
    }
}
